deployment/hasura/metadata/databases/tables/scheduler/scheduling_goal_definition.yaml

```diff
@@ -89,3 +92,20 @@ delete_permissions:
         {"_or":[
           {"author": {"_eq": "X-Hasura-User-Id"}},
           {"metadata":{"owner":{"_eq":"X-Hasura-User-Id"}}}]}
+
+event_triggers:
+  - definition:
+      enable_manual: false
+      insert:
+        columns: "*"
+      update:
+        columns:
+          - goal_id
+          - revision
+          - uploaded_jar_id
```

Comment:
Minor: None of the columns used for the `update` trigger are updatable.

Reply:
@mattdailis

Reply:
> None of the columns used for the update trigger are updatable

Sorry - just getting to this now. What does it mean for columns to not be updatable? Is that through permissions? Can an admin update these columns?

I believe this was inspired by the mission model trigger:
https://github.com/NASA-AMMOS/aerie/blob/3e4f4dbe5ecfc9bfd5d6356c6506ed4131e20e21/deployment/hasura/metadata/databases/tables/merlin/mission_model.yaml#L125-L127

Reply:
Yes, it is defined through the permissions, specifically the `update_permissions` key in the file. And no, an admin cannot update these later, because the `aerie_admin` role doesn't have these update permissions. Nor should they, as this is the `definition` table, where individual, write-once revisions live. A new jar would be provided via a new revision, at which point the `insert` trigger would handle the extraction.

Honestly, the `mission_model` on-update extraction should be removed as well, since even the `aerie_admin` cannot run it. I believe it's still around because before we decided on having a Gateway endpoint for rerunning extractions, it was going to be handled via an update trigger, and I only undid the changes made for the "Gateway-less retrigger" instead of properly removing it.

Thinking about this more, some future work could be to add another "trigger extraction" endpoint in the Gateway for these JARs in the event of a failure, similar to what was done for mission models.

Reply:
Commenting on this thread as it has yet to be resolved.

Reply:
Deleted the update section